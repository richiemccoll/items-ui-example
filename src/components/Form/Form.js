import React, { Component } from "react";
import { Button, Form } from "semantic-ui-react";
import PropTypes from 'prop-types';

class FormContainer extends Component {
  state = {
    file: "",
    result: ""
  };

  handleImageChange = e => {
    e.preventDefault();
    let fileReader = new FileReader();
    let file = e.target.files[0];

    fileReader.onloadend = () => {
      this.setState({ file: file, result: fileReader.result });
    };

    fileReader.readAsDataURL(file);
  };

  render() {
    const { action, inEdit, handleAction } = this.props;
    return (
      <Form>
        <Form.Field>
          <label>Title</label>
          <input
            placeholder="Title"
            ref={title => this.title = title}
            defaultValue={action === "edit" ? inEdit.title : ""}
          />
        </Form.Field>
        <Form.Field>
          <label>Description</label>
          <input
            ref={desc => this.desc = desc}
            placeholder="Description"
            defaultValue={action === "edit" ? inEdit.description : ""}
          />
        </Form.Field>
        <Form.Field>
          <input
            id="image"
            type="file"
            onChange={e => this.handleImageChange(e)}
          />
        </Form.Field>
        <Button
          onClick={e =>
            handleAction(e, {
              id: inEdit ? inEdit.id : 0,
              title: this.title.value,
              desc: this.desc.value,
              image: this.state.result || inEdit.image || null
            })}
        >
          Submit
        </Button>
      </Form>
    );
  }
}

export default FormContainer;

FormContainer.PropTypes = {
    action: PropTypes.string.isRequired,
    inEdit: PropTypes.object,
    handleAction: PropTypes.func.isRequired
}