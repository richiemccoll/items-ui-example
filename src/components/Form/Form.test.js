import React from "react";
import Form from "./Form";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import sinon from "sinon";
import chai from "chai";

describe("Form Component", () => {
  const formProps = {
    action: "add",
    inEdit: null,
    handleAction() {
      return;
    }
  };

  it("should render correctly based on props", () => {
    const component = renderer.create(
      <Form action={formProps.action} handleAction={formProps.handleAction} />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should call handleAction", () => {
    const handleAction = sinon.spy();
    const wrapper = mount(
      <Form
        inEdit={{ title: "test", description: "test", image: "" }}
        handleAction={handleAction}
      />
    );
    wrapper.find("button").simulate("click");
    chai.expect(handleAction).to.have.property("callCount", 1);
  });
});
