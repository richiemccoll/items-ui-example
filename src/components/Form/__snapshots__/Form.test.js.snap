// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Component should render correctly based on props 1`] = `
<form
  className="ui form"
>
  <div
    className="field"
  >
    <label>
      Title
    </label>
    <input
      defaultValue=""
      placeholder="Title"
    />
  </div>
  <div
    className="field"
  >
    <label>
      Description
    </label>
    <input
      defaultValue=""
      placeholder="Description"
    />
  </div>
  <div
    className="field"
  >
    <input
      id="image"
      onChange={[Function]}
      type="file"
    />
  </div>
  <button
    className="ui button"
    onClick={[Function]}
    tabIndex={undefined}
  >
    Submit
  </button>
</form>
`;
